internal/ansible/ansible_manager.go

```diff
@@ -0,0 +1,351 @@
+			}
+			log.Infof("ansible playbook execution completed of messageID %s", d.MessageId)
+			break loop
+
+		case results := <-playbookResults:
+			log.Debugf("posting events for messageID %s", d.MessageId)
+			err := a.sendEvents(results, reqFields.returnURL, responseTo, playbookYamlFile)
+			if err != nil {
+				log.Errorf("cannot post ansible playbook results of message %s: %v", d.MessageId, err)
+			}
+		}
+	}
+
+	if errRunPlaybook != nil {
+		// last event should be the failure, find the reason
+		msgList := a.ansibleDispatcher.GetMsgList()
+		errorCode, errorDetails := parseFailure(msgList[len(msgList)-1])
+		// required event for cloud connector
+		onFailed := a.ansibleDispatcher.ExecutorOnFailed(reqFields.crcDispatcherCorrelationID, "", errorCode, fmt.Sprintf("%v", errorDetails))
+		a.ansibleDispatcher.AddRunnerJobEvent(onFailed)
+	}
+	return nil
+}
+
+// sendEvents adds the events of AnsiblePlaybookJSONResults into eventList and sends them to the dispatcher.
+func (a *AnsibleManager) sendEvents(results *ansibleResults.AnsiblePlaybookJSONResults, returnURL string, responseTo string, playbookYamlFile string) error {
+	eventList := a.ansibleDispatcher.AddEvent(playbookYamlFile, results)
+	message, err := dispatcher.ComposeDispatcherMessage(eventList, returnURL, responseTo)
+	if err != nil {
+		log.Errorf("cannot compose message for events: %v. Error: %v", eventList, err)
+		return err
+	}
+	log.Infof("Message to be sent as reply: %v", string(message.Content))
+	_, err = a.dispatcherClient.Send(context.Background(), message)
+	if err != nil {
+		log.Errorf("cannot send message %s to the dispatcher. Content: %s. Error: %v", message.MessageId, string(message.Content), err)
+		return err
+	}
+	return nil
+}
+
+func (a *AnsibleManager) ExecutePreviousPlaybooks() error {
+	timeout := 300 * time.Second // Deafult timeout
+
+	// defined how to connect to hosts
+	ansiblePlaybookConnectionOptions := &options.AnsibleConnectionOptions{
+		Connection: "local",
+	}
+	// defined which should be the ansible-playbook execution behavior and where to find execution configuration.
+	ansiblePlaybookOptions := &playbook.AnsiblePlaybookOptions{
+		Inventory: "127.0.0.1,",
+	}
+
+	playbookCmd := &playbook.AnsiblePlaybookCmd{
+		ConnectionOptions: ansiblePlaybookConnectionOptions,
+		Options:           ansiblePlaybookOptions,
+	}
+
+	buffOut := new(bytes.Buffer)
+	buffErr := new(bytes.Buffer)
+
+	setupPlaybookCmd(playbookCmd, buffOut, buffErr)
+	allPlaybooks := a.MappingRepository.GetAll()
+	var errors error
+	for _, v := range allPlaybooks {
+		playbookCmd.Playbooks = []string{v}
+		res, err := a.execPlaybookSync(playbookCmd, timeout, buffOut, a.MappingRepository)
+		log.Error(err)
+		errors = multierror.Append(errors, err)
+		if res == nil && err != nil {
+			return err
+		}
+		buffOut.Reset()
+		buffErr.Reset()
+	}
+	return errors
+}
+
+func (a *AnsibleManager) StopPlaybooks() {
```

Comment:
I'm not sure about this Stop function, should be waitPlaybooks finish or something like that, no?